sw/device/lib/dif/autogen/dif_uart_autogen.inc

```diff
@@ -24,7 +24,7 @@ extern "C" {
 #endif  // __cplusplus
 
 /**
- * A handle to UART.
+ * A handle to uart.
```

Comment:
This is a regression I think?

Reply:
This makes life easier by only using ${ip.name_snake} values (or derivatives thereof) in the templates for the completely autogenerated DIF since this name can be parsed directly from the file name (the not-so-pretty line 167 you point out below). Normally this string is derived from the --peripheral name flag and passed in by the user of this script on its first invocation (the peripheral name is a longer name of the IP used for documentation, e.g. the semi-autogenerated markdown checklists). If I want to batch regenerate the completely autogenerated DIFs (nice for updating the autogenerated DIFs when the templates change) I would have to parse the peripheral name from the semi-autogenerated markdown checklist for an IP which would be even uglier. Hence I went with this option, i.e. remove it from the template all together,  though do let me know if you have a better idea/solution!